src/Middleware/BaseMiddleware.php

```diff
@@ -37,12 +37,12 @@ public function __construct(JWTAuth $auth)
      *
      * @param   \Illuminate\Http\Request $request
      *
-     * @throws \Symfony\Component\HttpKernel\Exception\BadRequestHttpException
+     * @throws \Tymon\JWTAuth\Exceptions\JWTException
      */
     public function checkForToken(Request $request)
     {
         if (! $this->auth->parser()->setRequest($request)->hasToken()) {
-            throw new BadRequestHttpException('Token not provided');
+            throw new JWTException('Token not provided');
```

Comment:
@tymondesigns I thought of that, that middle-wares should thrown HTTP exception. I'm trying to find a way to allow the application to intercept/localize the message though.
It would be a better approach to pass an exception code as the third argument to `BadRequestHttpException`?
Or add configuration to customize the message?

Or is up to the app code to detect this specific exception by comparing the message string:
```php
if ($e instanceof BadRequestHttpException && $e->getMessage() == 'Token not provideed') {
...
}
```
?